### ISSUE TITLE
Deserialize "$value" as `String` or `Vec<SomeStruct>` based on attributes

### ISSUE THREAD

**Issue by futile** [RReverser/serde-xml-rs#69]
I have data which can look like either:

```xml
<data encoding="XML">
  <tile gid="0"/>
  <tile gid="1"/>
</data>
```

or

```xml
<data encoding="base64">
  aaabbbbbbbcc
</data>
```

And, as far as I can see, this is not yet possible. With the old serde-xml I was able to get a `serde_xml::Value`, which was an enum that could be, e.g., `Value::Element(...)` or Value::Content(...)`, which made it possible to handle them individually. However, I have no Idea how I would do that currently.

The regular `Deserializer`/`Visitor` pattern seems to fail here, as I need to dispatch based on what is actually inside the XML, and don't known that beforehand.

Any help is appreciated!

**Comment by NuSkooler** [comment]
@futile did you find an answer to this?

**Comment by futile** [comment]
> @futile did you find an answer to this?

Nope, don't think I ever found a solution. But, given that the last commit to this repo was ~2 years ago, I would assume that there exist better options for XML-parsing (maybe without serde) for Rust by now. Good luck! :)

**Comment by NuSkooler** [comment]
@futile Thanks for your response on this! Sort of what I was afraid of... think I'll try quick_xml.

**Comment by punkstarman** [comment]
@futile , @NuSkooler , I've been the sole maintainer and the last 2 years were rough. I've been able to commit some more time this year.

Untagged enums would be difficult to support in XML. `deserialize_any` can't determine what type something is because XML is not self-describing.

I might implement a `Value` type in the future.
